guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/git/RepositoryBuilder.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+    */
+package org.guvnor.structure.backend.repositories.git;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.guvnor.structure.repositories.PublicURI;
+import org.guvnor.structure.repositories.Repository;
+import org.guvnor.structure.repositories.impl.DefaultPublicURI;
+import org.guvnor.structure.repositories.impl.git.GitRepository;
+import org.guvnor.structure.server.config.ConfigGroup;
+import org.guvnor.structure.server.config.ConfigItem;
+import org.guvnor.structure.server.config.PasswordService;
+import org.guvnor.structure.server.config.SecureConfigItem;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.file.FileSystemAlreadyExistsException;
+
+import static org.uberfire.backend.server.util.Paths.*;
+
+public class RepositoryBuilder {
```

Comment:
It is. I'm with you. We should name any Project to MavenProject or KieProject and so on.